src/master/flags.cpp

```diff
@@ -460,6 +460,18 @@ mesos::internal::master::Flags::Flags()
       "load an alternate allocator module using `--modules`.",
       DEFAULT_ALLOCATOR);
 
+  add(&Flags::hierarchical_recovery_factor,
+      "hierarchical_recovery_factor",
+      "Ratio of minimal re-registred agent before sending\n"
+      "offers after a leader re-election",
+      DEFAULT_AGENT_RECOVERY_FACTOR);
+
+  add(&Flags::hierarchical_recovery_timeout,
+      "hierarchical_recovery_timeout",
```

Comment:
Maybe `allocator_recovery_timeout`? 

In the master code that fills `allocator::Options` nothing implies that the created allocator is indeed _the_ hierarchical allocator. 
Moreover, the existence of recovery interval is not relevant to the fact that the allocator treats resource roles hierarchically.